src/RamRelation.h

```diff
@@ -314,6 +322,86 @@ class RamRelation {
         num_tuples++;
     }
 
+    /** insert a new tuple to table, possibly more than one tuple depending on relation type */
+    void insert(const RamDomain* tuple) {
+        // TODO: (pnappa) an eqrel check here is all that appears to be needed for implicit additions
+        if (id.isEqRel()) {
+            // TODO: future optimisation would require this as a member datatype
+            // brave soul required to pass this quest
+            // // specialisation for eqrel defs
+            // std::unique_ptr<binaryrelation> eqreltuples;
+            // in addition, it requires insert functions to insert into that, and functions
+            // which allow reading of stored values must be changed to accommodate.
+            // e.g. insert =>  eqRelTuples->insert(tuple[0], tuple[1]);
+
+            // for now, we just have a naive & extremely slow version, otherwise known as a O(n^2) insertion
+            // ):
+
+            // store all values that will be implicitly relevant to the two that we will insert
+            std::list<const RamDomain*> relevantStored;
```

Comment:
Vector more efficient for only iteration and appending?

Lists are O(1) appending, whilst vectors can carry a O(n) cost for the equivalent operation.

The only benefit I can see is that you may thrash memory allocations with a list, but that can be improved by just changing the std::list to a souffle::BlockList